vita3k/packages/src/sce_utils.cpp

```diff
@@ -1101,20 +833,253 @@ std::tuple<uint64_t, SelfType> get_key_type(std::ifstream &file, const SceHeader
     }
 }
 
-void decrypt_fself(const fs::path &file_path, KeyStore &SCE_KEYS, unsigned char *klictxt) {
-    auto np = file_path;
-    np.replace_extension(np.extension().string() + ".elf");
-    uint64_t authid = 0x2F00000000000001ULL;
-    self2elf(file_path, np, SCE_KEYS, klictxt, &authid);
-    fs::rename(np, file_path);
-    np = file_path;
-    np.replace_extension(np.extension().string() + ".fself");
-    make_fself(file_path, np, authid);
-    fs::rename(np, file_path);
-}
+std::vector<uint8_t> decrypt_fself(const std::vector<uint8_t> fself, const uint8_t *klic) {
+    const SCE_header &self_header = *reinterpret_cast<const SCE_header *>(fself.data());
+    const segment_info *const seg_infos = reinterpret_cast<const segment_info *>(&fself[self_header.section_info_offset]);
+    const AppInfoHeader app_info_hdr = AppInfoHeader((char *)(&fself[self_header.appinfo_offset]));
+    const auto is_app = app_info_hdr.self_type == SelfType::APP;
+
+    // Check if the self is encrypted
+    if (seg_infos->encryption != 2) {
+        // Check if the self is an app and if a klic have all 0 inside it
+        if (is_app && std::all_of(klic, klic + 16, [](uint8_t i) { return i == 0; })) {
+            LOG_ERROR("No klic provided for encrypted App");
+            return {};
+        }
+
+        // Get the keyset for the self
+        KeyStore SCE_KEYS;
+        register_keys(SCE_KEYS, 1);
+        uint64_t authid = 0x2F00000000000001ULL;
+
+        // Extract the elf from the self
+        std::vector<char> elf;
```

Comment:
```suggestion
        std::vector<uint8_t> elf;
```
No reason to use a different type compared to fself (also uint8_t is for raw data, char is for strings)

Reply:
For it i no can change like that, relate two using after, no works with uint8_t exept with reinterpret, i have no touched it, is like before

Reply:
Done all